compiler/rustc_resolve/src/late/diagnostics.rs

```diff
@@ -1966,13 +1966,22 @@ impl<'a: 'ast, 'ast, 'tcx> LateResolutionVisitor<'a, '_, 'ast, 'tcx> {
     fn let_binding_suggestion(&mut self, err: &mut Diagnostic, ident_span: Span) -> bool {
         if let Some(Expr { kind: ExprKind::Assign(lhs, ..), .. }) =
             self.diagnostic_metadata.in_assignment
-            && let ast::ExprKind::Path(None, _) = lhs.kind
+            && let ast::ExprKind::Path(None, ref path) = lhs.kind
         {
             if !ident_span.from_expansion() {
+                let (span, text) = match path.segments.first().map(|it| it.ident.as_str()) {
+                    Some(name) if let Some(name) = name.strip_prefix("let") => {
```

Comment:
```suggestion
                    Some(seg) if let Some(name) = seg.ident.as_str().strip_prefix("let") => {
```
To avoid an unnecessary `map`.